clouddriver-azure/src/main/groovy/com/netflix/spinnaker/clouddriver/azure/client/AzureNetworkClient.groovy

```diff
@@ -272,68 +377,38 @@ class AzureNetworkClient extends AzureBaseClient {
+      def vnets = this.client.getVirtualNetworksOperations().listAll().body
+      def currentTime = System.currentTimeMillis()
+      vnets.each { item->
         if (item.location == region) {
-          getSubnetForVirtualNetwork(item).each { AzureSubnetDescription subnet -> result += subnet }
+          getSubnetForVirtualNetwork(item).each { AzureSubnetDescription subnet ->
+            subnet.lastReadTime = currentTime
+            result += subnet
+          }
         }
       }
     } catch (Exception e) {
       log.info("getSubnetsAll -> Unexpected exception " + e.toString())
     }
-
     result
   }
 
   /**
-   * Retrieve a collection of all subnets for a give set of credentials, regardless of region, optionally
-   * filtered for a given resource group
-   * @param creds the credentials to use when communicating to the Azure subscription(s)
-   * @param resourceGroupName specify the resource group that is used to filter subnets; only
-   *
-   * @return a Collection of objects which represent a Subnet in Azure
-   */
-  Collection<AzureSubnetDescription> getSubnetsInResourceGroup(AzureCredentials creds, String resourceGroupName) {
-
-    def result = new ArrayList<AzureSubnetDescription>()
-
-    try {
-      this.getNetworkResourceProviderClient(creds).getVirtualNetworksOperations().list(resourceGroupName).virtualNetworks.each { item->
-        getSubnetForVirtualNetwork(item).each { AzureSubnetDescription subnet -> result += subnet }
-      }
-    } catch (Exception e) {
-      log.info("getSubnetsAll -> Unexpected exception " + e.toString())
-    }
-
-    result
-  }
-
-  /**
-   * Retrieves a particular subnet from a given resource group based on its name
-   * @param creds
-   * @param resourceGroupName
-   * @param subnetName
-     * @return an AzureSubnetDescription instance containing details about the given subnet
-     */
-  AzureSubnetDescription getSubnet(AzureCredentials creds, String resourceGroupName, String subnetName) {
-    getSubnetsInResourceGroup(creds, resourceGroupName).find {it.name == subnetName}
-  }
-
-  /**
    * Gets a virtual network object instance by name, or null if the virtual network does not exist
    * @param creds the credentials to use when communicating with Azure subscription(s)
    * @param resourceGroupName name of the resource group to look in for a virtual network
    * @param virtualNetworkName name of the virtual network to get
    * @return virtual network instance, or null if it does not exist
    */
-  VirtualNetwork getVirtualNetwork(AzureCredentials creds, String resourceGroupName, String virtualNetworkName) {
-    this.getNetworkResourceProviderClient(creds).
-      getVirtualNetworksOperations().
-      get(resourceGroupName, virtualNetworkName).
-      getVirtualNetwork()
+  VirtualNetwork getVirtualNetwork(String resourceGroupName, String virtualNetworkName) throws CloudException, IOException, IllegalArgumentException {
+    client
```

Comment:
There is an instance of it being used in AzureResourceManagerClient. I've also seen this convention used in other places outside of our CloudDriver code (e.g., GCEUtils)  That said, I'm not sure it's necessary. If I remove them, the code still compiles. I'll remove for now and we can monitor it.